client/my-sites/no-results/index.jsx

```diff
@@ -3,27 +3,25 @@
  */
 import React from 'react';
 
+/**
+ * Internal dependencies
+ */
+import { useTranslate } from 'i18n-calypso';
+
 /**
  * Style dependencies
  */
 import './style.scss';
 
-class NoResults extends React.Component {
-	static defaultProps = {
-		text: 'No results',
-		image: false,
-	};
-
-	render() {
-		return (
-			<div className="no-results">
-				{ this.props.image ? (
-					<img className="no-results__img" src={ this.props.image } alt="" />
-				) : null }
-				<span>{ this.props.text }</span>
-			</div>
-		);
+export default function NoResults( { text, image = false } ) {
+	const translate = useTranslate();
+	if ( text === undefined ) {
+		text = translate( 'No results.' );
```

Comment:
Here's an approach that I think is more elegant:
```js
function NoResults( props ) {
  const translate = useTranslate();
  const { image, text = translate( 'No results.' ) } = props;
  return <jsx>...</jsx>;
}
```
I.e., moving the destructuring into the function body allows us to use `translate` to create the default value.